### PR TITLE
Don't report parse errors for modifiers on event accessors.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
This PR specifically makes it so that modifiers on event accessor are not checked (while parsing). Previously, we disallowed (at parse time) certain modifier combinations. There's no reason for this complexity and specificity to be at the parser level. Syntactically we know what's going on, and we can allow this while parsing and check and report the error later. It also prevents incrementally reusing parameters (and their containers) when this happens.

Tagging @dotnet/roslyn-compiler @gafter @VSadov

This part of a long term effort to remove unnecessary errors from the parser and move them to later stages of the compiler pipeline. Doing this makes the parser simpler, makes it faster to get a syntax tree, and makes our incremental implementation simpler (because the parser/incremental-parser needs to track less contextual states).

This is also important as syntax-diagnostics end up causing issues for higher level IDE features.

File: src/Compilers/CSharp/Test/Syntax/Parsing/ParserErrorMessageTests.cs [hunk 1]
```diff
@@ -3878,8 +3878,9 @@ public static int Main()
     }
 }
 ";
-
-            ParseAndValidate(test, Diagnostic(ErrorCode.ERR_NoModifiersOnAccessor, "private"), Diagnostic(ErrorCode.ERR_NoModifiersOnAccessor, "public"));
+            CreateCompilationWithMscorlib(test).GetDeclarationDiagnostics().Verify(
+                Diagnostic(ErrorCode.ERR_NoModifiersOnAccessor, "private"),
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Syntax/Parsing/ParserErrorMessageTests.cs]: >Diagnostic(ErrorCode.ERR_NoModifiersOnAccessor, "private") [](start = 16, length = 58)

Please include comments with error message and the actual line of code for the error.

File: src/Compilers/CSharp/Test/Syntax/Parsing/ParserErrorMessageTests.cs [hunk 2]
```diff
@@ -3897,7 +3898,9 @@ public class Test
 }
 ";
 
-            ParseAndValidate(test, Diagnostic(ErrorCode.ERR_NoModifiersOnAccessor, "public"), Diagnostic(ErrorCode.ERR_NoModifiersOnAccessor, "private"));
+            CreateCompilationWithMscorlib(test).GetDeclarationDiagnostics().Verify(
+                Diagnostic(ErrorCode.ERR_NoModifiersOnAccessor, "public"),
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Syntax/Parsing/ParserErrorMessageTests.cs]: >Diagnostic(ErrorCode.ERR_NoModifiersOnAccessor, "public"), [](start = 16, length = 58)

Same here.

**Comment by AlekseyTs** [comment]
LGTM, modulo the missing comment for diagnostics in test.

File: src/Compilers/CSharp/Test/Syntax/Parsing/ParserErrorMessageTests.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Test/Syntax/Parsing/ParserErrorMessageTests.cs]: sure

File: src/Compilers/CSharp/Test/Syntax/Parsing/ParserErrorMessageTests.cs (hunk 2 above)
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Test/Syntax/Parsing/ParserErrorMessageTests.cs]: Done.

**Pull request merged**